api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,317 @@
+			Base: shared.TLSConfigFile{
+				TLSRootCAFile: os.Getenv("SERVER_TLS_ROOT_CA_FILE"),
+			},
+		},
+	}
+	workflowName := "probe-workflow"
+	var wfrId *string
+	go func() {
+		wfrId, cleanup, err = m.run(cancellableContext, cf, workflowName, events, errorChan)
+		if err != nil {
+			errorChan <- err
+		}
+
+		// check if the context was cancelled in the meantime
+
+		if cancellableContext.Err() != nil {
+			cleanupErr := cleanup()
+			if cleanupErr != nil {
+				m.l.Error().Msgf("error cleaning up probe worker: %s", cleanupErr)
+			}
+		}
+	}()
+
+	defer func() {
+		var cleanupErr error
+		if cleanup != nil {
+			cleanupErr = cleanup()
+		} else {
+			m.l.Warn().Msg("cleanup function was never set for probe")
+		}
+		if cleanupErr != nil {
+			m.l.Error().Msgf("error cleaning up probe worker: %s", err)
+		}
+	}()
+
+	defer func() {
+
+		time.Sleep(250 * time.Millisecond)
+
+		if wfrId != nil && *wfrId != "" {
+
+			wrfRow, err := m.config.APIRepository.WorkflowRun().GetWorkflowRunById(context.Background(), cf.TenantId, *wfrId)
+
+			if err != nil {
+				m.l.Error().Msgf("error getting workflow run: %s", err)
+
+			}
+
+			workflowId := sqlchelpers.UUIDToStr(wrfRow.Workflow.ID)
+
+			_, err = m.config.APIRepository.Workflow().DeleteWorkflow(cf.TenantId, workflowId)
+
+			m.l.Info().Msgf("deleted workflow %s", workflowId)
+			if err != nil {
+				m.l.Error().Msgf("error deleting workflow: %s", err)
+			}
+		} else {
+			m.l.Warn().Msg("workflow run id was never set")
+		}
+
+	}()
+
+	messages := []string{"This is a stream event for step-one", "This is a stream event for step-two"}
+	messageIndex := 0
+
+	for {
+
+		select {
+		case e := <-events:
+			if e != messages[messageIndex] {
+				return nil, fmt.Errorf("expected message %s, got %s", messages[messageIndex], e)
+			}
+
+			messageIndex++
+			if messageIndex == len(messages) {
+				m.l.Debug().Msg("probe completed successfully")
+				return gen.MonitoringPostRunProbe200JSONResponse{}, nil
+			}
+
+		case <-cancellableContext.Done():
+
+			if cancellableContext.Err() == context.DeadlineExceeded {
+				m.l.Error().Msg("timed out waiting for probe to complete")
+				return nil, fmt.Errorf("timed out waiting for probe to complete")
+			} else {
+				m.l.Debug().Msg("probe completed successfully")
+				return gen.MonitoringPostRunProbe200JSONResponse{}, nil
+			}
+
+		case err := <-errorChan:
+			m.l.Error().Msgf("error during probe: %s", err)
+			return nil, err
+		case <-interrupt:
+			return nil, fmt.Errorf("interrupted during probe")
+		}
+
+	}
+
+}
+
+type probeEvent struct {
+	UniqueStreamId string
+}
+
+type stepOneOutput struct {
+	Message        string `json:"message"`
+	UniqueStreamId string
+}
+
+func (m *MonitoringService) run(ctx context.Context, cf clientconfig.ClientConfigFile, workflowName string, events chan<- string, errors chan<- error) (*string, func() error, error) {
+
+	c, err := client.NewFromConfigFile(
+		&cf, client.WithLogLevel("info"),
```

Comment:
Before merging, we might want to change this log level to match the server, otherwise this will be very noisy